packages/frontend/src/hooks/useTableConfig.ts

```diff
@@ -1,27 +1,106 @@
-import { ApiQueryResults, Explore, TableChart } from '@lightdash/common';
+import {
+    AdditionalMetric,
+    ApiQueryResults,
+    ColumnProperties,
+    Explore,
+    Field,
+    friendlyName,
+    getAdditionalMetricLabel,
+    getItemLabel,
+    getItemMap,
+    isAdditionalMetric,
+    isField,
+    TableCalculation,
+    TableChart,
+} from '@lightdash/common';
 import { useMemo, useState } from 'react';
 
 const useTableConfig = (
     tableChartConfig: TableChart | undefined,
     resultsData: ApiQueryResults | undefined,
     explore: Explore | undefined,
+    columnOrder: string[],
 ) => {
     const [showTableNames, setShowTableName] = useState<boolean>(
         tableChartConfig?.showTableNames === undefined
             ? true
             : tableChartConfig.showTableNames,
     );
 
+    const [columnProperties, setColumnProperties] = useState<
+        ColumnProperties[]
+    >(tableChartConfig?.columns === undefined ? [] : tableChartConfig?.columns);
+
+    const itemMap = useMemo<
+        Record<string, Field | AdditionalMetric | TableCalculation>
+    >(() => {
+        if (explore && resultsData) {
+            return getItemMap(
+                explore,
+                resultsData.metricQuery.additionalMetrics,
+                resultsData.metricQuery.tableCalculations,
+            );
+        }
+        return {};
+    }, [explore, resultsData]);
+
+    const columnHeader = (fieldId: string) => {
+        const field = itemMap && itemMap[fieldId];
+        if (isAdditionalMetric(field)) {
+            // AdditionalMetric
+            return getAdditionalMetricLabel(field);
+        } else if (isField(field)) {
+            // Field
+            return showTableNames ? getItemLabel(field) : field.label;
+        } else {
+            //TableCalculation
+            return friendlyName(fieldId);
+        }
+    };
+
+    const filterVisible = (fieldId: string) =>
```

Comment:
```suggestion
    const isFilterVisible = (fieldId: string) =>
```

If we switch columns to be a map instead of an array we can get rid of this function. We could have `columns[fieldId]?.visible ?? true`